routes.py

```diff
@@ -369,24 +379,53 @@ def createAccount():
 @app.route("/deleteAccount", methods=["POST"])
 @login_required
 def deleteAccount():
-    userID = flask.request.args.get("userID")
+    userID = current_user.id
     print("Printing UserID")
     print(userID)
-    delUser = user.query.filter_by(user.id == userID).first()
+
+    # get all items in favorite restaurants table with the userID
+    delRestaurants = favorite_restraunts.query.filter_by(user_id=userID).all()
+    for restaurant in delRestaurants:
+        db.session.delete(restaurant)
+        db.session.commit()
+
+    # get all items in friends table with the userID
+    delFriends = friends.query.filter_by(user_id=userID).all()
+    for friend in delFriends:
+        db.session.delete(friend)
+        db.session.commit()
+
+    # get all items in posts table with the userID
+    delPosts = user_post.query.filter_by(user_id=userID).all()
+    for post in delPosts:
+        # get the comments for this post and delete all of them
+        delComments = post_comments.query.filter_by(post_id=post.id).all()
+        for comment in delComments:
+            db.session.delete(comment)
+            db.session.commit()
+
+        db.session.delete(post)
+        db.session.commit()
+
+    # delete the user
+    delUser = user.query.filter_by(id=userID).first()
     db.session.delete(delUser)
     db.session.commit()
 
-    status = "success"
-    if user.query.filter_by(user.id == userID).first():
-        status = "failed"
+    status = 200
+    if user.query.filter_by(id=userID).first():
+        status = 400
+
+    # log user out after deleting account
+    logout_user()
+    flask.session.pop("google_token", None)
+
     return {"status": status}
 
 
-@app.route("/createPost", methods=["POST", "GET"])
+@app.route("/createPost", methods=["POST"])
```

Comment:
I believe createPosts does need GET included at the top because when a request is made to this endpoint it is a GET request instead of a POST